### PR TITLE
Hotfix/Title case button

### PR DESCRIPTION
**Pull request by VirginiaDooley**
Closes https://github.com/DemocracyClub/yournextrepresentative/issues/1480

https://user-images.githubusercontent.com/7017118/140958598-86150e78-d099-4f2a-8158-b96e87841e7b.mp4

### CODE REVIEW

**Comment by VirginiaDooley** [comment]
Thanks! Yes, I realised all this as soon as I pushed :) Instead, I did something similar to what we have with SOPN parsing `clean_name` before saving the form (either for a new person or a person update). Happy to replace that with your javascript suggestion, but what I've done works without any user input.

File: ynr/apps/candidates/static/js/person_form.js [hunk 1]
```diff
@@ -194,3 +195,16 @@ function makeTitleCase() {
     return title_case
   })
 }
+
+function addTitleCaseButton() {
+  let btn = document.createElement("button");
+  btn.innerHTML = "Title Case";
+  btn.type = "submit";
```
Comment (symroe) [inline ynr/apps/candidates/static/js/person_form.js:199]: If you change this from "submit" to "button" then you should stop that validation error / jump you see in the video. A submit button tells the browser to submit the form (causing validation errors), whereas a "button" does nothing unless it's hooked up (like you do below)

File: ynr/apps/candidates/static/js/person_form.js [hunk 2]
```diff
@@ -194,3 +195,16 @@ function makeTitleCase() {
...
+  btn.className = "titleCaseNameField button tiny secondary"
+  document.querySelector(".person_name").insertAdjacentElement('afterend', btn);
+  btn.addEventListener("click", function () {
+    makeTitleCase()
+  });
+}
+
+document.body.insertAdjacentHTML( 'afterbegin', '<div id="myID">...</div>' );
```
Comment (symroe) [inline ynr/apps/candidates/static/js/person_form.js]: Is this left over testing code?

**Review by symroe — CHANGES_REQUESTED** [review]
I think this still needs work to support more than one name field per page.

There are some pages that have more than one name field on, and in that case I think this code will just attach a button to the first instance.

I tested this by altering the bulk adding form to add the class, like this (this is actually a change that's work making!)

```diff
diff --git ynr/apps/bulk_adding/forms.py ynr/apps/bulk_adding/forms.py
index 2c8da542..9fc1bc92 100644
--- ynr/apps/bulk_adding/forms.py
+++ ynr/apps/bulk_adding/forms.py
@@ -165,7 +165,8 @@ class BaseBulkAddReviewFormSet(BaseBulkAddFormSet):
 
 class NameOnlyPersonForm(forms.Form):
     name = forms.CharField(
-        label="Name (style: Ali Smith, not SMITH Ali)", required=True
+        label="Name (style: Ali Smith, not SMITH Ali)", required=True,
+        widget=forms.TextInput(attrs={"class": "person_name"})
     )
     ballot = ValidBallotField(
         widget=BallotInputWidget(attrs={"type": "hidden"})
```

I think the code needs to loop over each element with the class, create a new button for each, and append that button. There might be a better way of doing this, but it'll be more or less like that.

You can see the URL I'm looking at by going to `/bulk_adding/sopn/local.adur.buckingham.2021-05-06/` (or any valid ballot paper ID)

File: ynr/apps/candidates/static/js/person_form.js (hunk 2 above)
Comment (VirginiaDooley) [reply ynr/apps/candidates/static/js/person_form.js]: Yes

**Pull request merged**